src/plugins/controls/server/control_types/options_list/options_list_queries.ts

```diff
@@ -0,0 +1,174 @@
+  if (textFieldName && fieldSpec?.aggregatable && searchString) {
+    return suggestionAggSubtypes.keywordAndText;
+  }
+  if (fieldSpec?.type === 'boolean') {
+    return suggestionAggSubtypes.boolean;
+  }
+  if (fieldSpec && getFieldSubtypeNested(fieldSpec)) {
+    return suggestionAggSubtypes.subtypeNested;
+  }
+  return suggestionAggSubtypes.keywordOnly;
+};
+
+const getEscapedQuery = (q: string = '') =>
+  q.replace(/[.?+*|{}[\]()"\\#@&<>~]/g, (match) => `\\${match}`);
+
+const suggestionAggSubtypes: { [key: string]: OptionsListAggregationBuilder } = {
+  /**
+   * the "Keyword only" query / parser should be used when the options list is built on a field which has only keyword mappings.
+   */
+  keywordOnly: {
+    buildAggregation: ({ fieldName, searchString }: OptionsListRequestBody) => ({
+      terms: {
+        field: fieldName,
+        include: `${getEscapedQuery(searchString)}.*`,
+        execution_hint: 'map',
+        shard_size: 10,
+      },
+    }),
+    parse: (rawEsResult) =>
+      get(rawEsResult, 'aggregations.suggestions.buckets')?.map(
+        (suggestion: { key: string }) => suggestion.key
+      ),
+  },
+
+  /**
+   * the "Keyword and text" query / parser should be used when the options list is built on a multi-field which has both keyword and text mappings. It supports case-insensitive searching
+   */
+  keywordAndText: {
+    buildAggregation: (req: OptionsListRequestBody) => {
+      if (!req.textFieldName) {
+        // if there is no textFieldName specified, or if there is no search string yet fall back to keywordOnly
+        return suggestionAggSubtypes.keywordOnly.buildAggregation(req);
+      }
+      const { fieldName, searchString, textFieldName } = req;
+      return {
+        filter: {
+          match_phrase_prefix: {
+            [textFieldName]: getEscapedQuery(searchString),
+          },
+        },
+        aggs: {
+          keywordSuggestions: {
+            terms: {
+              field: fieldName,
+              shard_size: 10,
+            },
+          },
+        },
+      };
+    },
+    parse: (rawEsResult) =>
+      get(rawEsResult, 'aggregations.suggestions.keywordSuggestions.buckets')?.map(
+        (suggestion: { key: string }) => suggestion.key
+      ),
+  },
+
+  /**
+   * the "Boolean" query / parser should be used when the options list is built on a field of type boolean. The query is slightly different than a keyword query.
+   */
+  boolean: {
+    buildAggregation: ({ fieldName }: OptionsListRequestBody) => ({
+      terms: {
+        field: fieldName,
+        execution_hint: 'map',
+        shard_size: 10,
+      },
+    }),
+    parse: (rawEsResult) =>
+      get(rawEsResult, 'aggregations.suggestions.buckets')?.map(
+        (suggestion: { key_as_string: string }) => suggestion.key_as_string
+      ),
+  },
+
+  /**
+   * the "Subtype Nested" query / parser should be used when the options list is built on a field with subtype nested.
+   */
+  subtypeNested: {
```

Comment:
To test this, I created a tiny index following the [Elastic documentation on nested types](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html) and created a control from that - specifically, for `user.first.keyword`. This **wasn't** case sensitive even though I think it has the potential to be since `user.first` has the following mapping: 
```
... "first": {
  "type": "text",
  "fields": {
    "keyword": {
      "type": "keyword",
      "ignore_above": 256
    }
  }
}, ...
```
Not sure if it is worth exploring how to make these searches case sensitive or not? It's a pretty specific case so might not be worth it :)

Reply:
This is something I didn't test very thoroughly, so thank you for looking into it!

I think that making case-insensitive search work correctly with nested fields is something we'll have to look into at a later date, because I'm not exactly sure what was going wrong here to make it not work. It might end up being a wild goose chase!

Reply:
Just to be completely sure, when you tested this, the suggestions were coming up properly, just case sensitive, correct?

Reply:
Correct! I was able to get suggested values, but the search was case sensitive